src/librustc/hir/map/mod.rs

```diff
@@ -566,6 +619,13 @@ impl<'hir> Map<'hir> {
         self.find(id).unwrap_or_else(|| bug!("couldn't find node id {} in the AST map", id))
     }
 
+    // FIXME(@ljedrz): replace the NodeId variant
+    pub fn get_by_hir_id(&self, id: HirId) -> Node<'hir> {
+        // read recorded by `find`
+        self.find_by_hir_id(id).unwrap_or_else(||
+            bug!("couldn't find id {:?} in the HIR map", id))
+    }
```

Comment:
This should just convert the `HirId` to a `NodeId` and call the regular `get` method.